### DISCUSSION TITLE
Using urql with Next.js 13.2

### DISCUSSION THREAD

**Discussion by NgoranA** [urql-graphql/urql#3080] [Q&A] [answered]
Hello everyone. Please how can I integrate **urql** with the new **Next.js 13.2**?
 So far wherever I needed to use graphql I turned the component into a client component with "use-client" and imported into a server component but i want this to be done on the server.

**Comment by hgiasac** [comment, marked as answer]
The `urql` package is tightly coupled with React. The library initializes a [React Context](https://github.com/urql-graphql/urql/blob/f182601c5ae4b1ff2e492fc0dbc2f12621c2dbe4/packages/react-urql/src/context.ts#L12) at startup that causes exceptions on the server side. You must create the client from the `@urql/core` package in the server component instead. Avoid importing `urql` in the file.

```ts
import {
  Client,
  createClient,
  dedupExchange,
  fetchExchange,
} from "@urql/core";

const client = createClient({
    URL: "https://example.com/v1/graphql",
    exchanges: [dedupExchange, fetchExchange],
});

const result = await client.query(GetUsersDocument).toPromise();
```

**Reply by EfstathiadisD** [reply]
I am wondering. Just thinking out loud here. Since all these packages use context how much work would it be to not use it and use another Store Instance? That's not tied to React. CacheStorage should be available in both environments.

Of course we need to consider the React lifecycle sync which wouldn't happen automatically but am I off? Am I missing something major here?
